code/game/gamemodes/nuclear/nuclearbomb.dm

```diff
@@ -23,18 +23,17 @@ var/obj/item/weapon/disk/nuclear/nukedisk
 
 /obj/machinery/nuclearbomb/New()
 	..()
-	r_code = "[rand(10000, 99999.0)]"//Creates a random code upon object spawn.
+	r_code = "[rand(10000, 99999)]"//Creates a random code upon object spawn.
 
 /obj/machinery/nuclearbomb/process()
-	if (src.timing)
+	if(timing)
 		bomb_set = 1 //So long as there is one nuke timing, it means one nuke is armed.
-		src.timeleft--
-		if (src.timeleft <= 0)
+		timeleft -= SS_WAIT_MACHINERY/10 //Machinery does NOT tick every 10 ms. Divided by ten to convert into seconds
```

Comment:
Is it possible to move the whole ticking-down from process to fast process instead of cannibalizing process so it properly deducts?

Reply:
Possible, but it would refresh every 0.7 seconds instead of every 2 seconds. I'm not on a level where I can tell if that's a smart idea or not. Normally fast machinery is reserved for machines that need to do stuff faster than one every two seconds like conveyors specifically

Reply:
One single atom has a negligible impact on the subsystem as a whole, but this entire file could use a rewrite

Reply:
I'd prefer you used the SECONDS define

`SS_WAIT_MACHINERY / 1 SECONDS`